diff --git a/wasm/src/lib.rs b/wasm/src/lib.rs
--- a/wasm/src/lib.rs
+++ b/wasm/src/lib.rs
@@ -1,3 +1 @@
-use std::{sync::mpsc, thread, time};
-
 use quasi::{executor, lexer, parser, Error};
@@ -6,14 +4,4 @@
 #[wasm_bindgen]
-pub fn execute(input: &[u8], timeout_ms: u64) -> Result<String, String> {
-	let timeout = time::Duration::from_millis(timeout_ms);
-	let (tx, rx) = mpsc::channel();
-
-	thread::scope(move |s| {
-		s.spawn(move || s.spawn(move || tx.send(wrap(input).map_err(|e| e.format(input)))));
-	});
-
-	rx.recv_timeout(timeout)
-		.unwrap_or_else(|_| Err(format!("execution timed out after {}ms", timeout_ms)))
+pub fn execute(input: &[u8], _timeout_ms: u64) -> Result<String, String> {
+	wrap(input).map_err(|error| error.format(input))
 }
